src-docs/src/components/guide_section/guide_section.js

```diff
@@ -433,14 +435,37 @@ export class GuideSection extends Component {
     );
   }
 
+  componentDidUpdate() {
+    if (this.state.selectedTab.name === 'javascript') {
+      this.refs.javascript.childNodes[0].children[0].scrollTop = this.memoScroll;
+    }
+  }
+
   renderCode(name) {
-    return (
-      <div key={name} ref={name}>
-        <EuiCodeBlock language={nameToCodeClassMap[name]} overflowHeight={400}>
-          {this.state.renderedCode}
-        </EuiCodeBlock>
-      </div>
+    const euiCodeBlock = (
+      <EuiCodeBlock language={nameToCodeClassMap[name]} overflowHeight={400}>
+        {this.state.renderedCode}
+      </EuiCodeBlock>
     );
+
+    const divProps = {
+      key: name,
+      ref: name,
+    };
+
+    const memoScrollUtility = () => {
+      this.memoScroll = this.refs.javascript.childNodes[0].children[0].scrollTop;
```

Comment:
```suggestion
      const pre = this.refs.javascript.querySelector('.euiCodeBlock__pre');
      this.memoScroll = pre ? pre.scrollTop : 0;
```

There's not a great way to target the `EuiCodeBlock` `pre` tag, so this will at least guard agaist type errors.